### PR TITLE
V3 HiddenServices and Key Management

### PR DESCRIPTION
**Pull request by GuillaumeHubR**
# Proposed Changes

Integrate latest Tor Version and permit V3 Hiddenservices

## Related Issues

Hidden services V3 not working and Private key was build each restart

### CODE REVIEW

**Comment by GuillaumeHubR** [comment]
Passing V11 images, permit Private key management and HiddenServices in Tor V3 with Tor 0.4.6.9 release

**Comment by sveip** [comment]
@Psykopear is this verified, and can be merged so that https://github.com/hassio-addons/addon-tor/pull/105 also can move forward?

**Comment by Psykopear** [comment]
Hi all, sorry for the late reply.

I merged the origin's main into this branch so we have an updated base, than modified this pr's base branch to `fix-issue-104` on this fork.

I'm going to make a review now

File: tor/Dockerfile [hunk 1]
```diff
@@ -12,9 +12,9 @@ COPY rootfs /
 # generation and coreutils for base64/base32
 # commands
 RUN apk add --no-cache \
-        tor=0.4.6.7-r0 \
-        openssl=3.0.0-r2 \
-        coreutils=9.0-r1
+        tor \
+        openssl \
+        coreutils
```
Comment (Psykopear) [inline tor/Dockerfile]: I think we should follow frenk's advice here and pin dependencies:

https://github.com/hassio-addons/addon-tor/pull/105#issuecomment-944891247

**Comment by Psykopear** [comment]
Other than the pin on the dependencies it looks fine, once that is fixed I'll merge it.

**Comment by sveip** [comment]
@Psykopear, @GuillaumeHubR. It seems there isn't much drive to merge the last PR into HA, is it an option to move a fork to HACS?

**Comment by Psykopear** [comment]
Hi @sveip 

Being an official addon, I think we should try to have the fix merged upstream anyway.
I don't think the maintainers don't want to merge the fix, it just seems they did not have the time to review or did not notice the activity on the PR.
I'm open to work more on the PR if something needs to be done, but I'd like to have some feedback from them first.

That said, if @GuillaumeHubR is ok with it too, feel free to fork this repo, merge the fix in the main branch and do what's needed (if anything) to make it work with HACS.

In that case I'd add a note in the README explaining that it's supposed to be a temporary solution until the fix gets merged upstream, or until we receive a feedback from the maintainers about the future plans for the addon.

**Comment by sveip** [comment]
@Psykopear I agree on most points, I'm def. not saying it's intentional overlooked by the maintainer. The argument I'm making is that because of the lack of focus (for whichever valid reason) from the maintainer of HA, it might be better for all parties to not have it as an official addon.

**Comment by Psykopear** [comment]
Yes, currently the official addon is not usable, so if this situation does not change it's surely better to have a working one maintained by the community than an official broken one.

What I wanted to say is that I can't commit to the maintainance of this addon if it becomes a community one, but if anyone wants to, they can use this fix without problems.

I still think that after this fix the addon should not break for a while, so it's worth to try to contact the maintainers again and see if we can get this merged upstream.
I might try to send an email later today to see if I can get their attention.

I know, it can be frustrating, but unless we plan to add more features and do things that are not in the scope of the official addon, I feel like contributing to the upstream project is the best thing to do.

PS: It might make sense to make the fix available in HACS to let people test the new version and also to have an alternative until the official addon is either fixed or removed

**Comment by sveip** [comment]
I totally agree on all points. Thanks for contacting them. I've quietly comment on the PR twice during the last few months without any success, so I didn't want to bug them yet another time as it might be counter-productive/demotivational.

**Comment by sveip** [comment]
Hi @Psykopear, I cloned this repo to my `/root/addons`, checked out this branch `fix-issue-104` and tried installing from HA (`http://192.168.8.123:8123/hassio/store`). 

The first time I clicked install I got an error and it failed, but the second time it succeeded. I didn't copy the error message unfortunately.

**Pull request merged**